habitat-lab/habitat/tasks/rearrange/utils.py

```diff
@@ -431,7 +437,53 @@ def place_robot_at_closest_point(
             f"Object {target_position} is out of bounds but trying to set robot position to {agent_pos}"
         )
 
+    # Get the normal agent position
+    agent_pos = sim.safe_snap_point(target_position)
     desired_angle = get_angle_to_pos(np.array(target_position - agent_pos))
+    # Cache the initial location of the agent
+    cache_pos = sim.articulated_agent.base_pos
+
+    accept_agent_center = []
+
+    if navmesh_offset is not None:
+        # Set the base pos of the agent
+        sim.articulated_agent.base_pos = agent_pos
+        # Project the nav pos
+        nav_pos_3d = [
+            np.array([xz[0], cache_pos[1], xz[1]]) for xz in navmesh_offset
+        ]
+        # Do transformation to get the location
+        center_pos_list = [
+            sim.articulated_agent.sim_obj.transformation.transform_point(xyz)
+            for xyz in nav_pos_3d
+        ]
+        for center_pos in center_pos_list:
+            # Update the transformation of the agent
+            sim.articulated_agent.base_pos = center_pos
+            # Get the transformation
+            trans = sim.articulated_agent.sim_obj.transformation
+            cur_pos = [trans.transform_point(xyz) for xyz in nav_pos_3d]
+            # Project the height
+            cur_pos = [
+                np.array([xz[0], cache_pos[1], xz[2]]) for xz in cur_pos
+            ]
+            is_navigable = [
+                sim.pathfinder.is_navigable(pos) for pos in cur_pos
```

Comment:
Again, it would be nice to early out on the first failed `pathfinder.is_navigable`.

Reply:
thanks!